src/Runner.Worker/action_yaml.json

```diff
@@ -120,7 +120,22 @@
                 "hashFiles(1,255)"
             ],
             "sequence": {
-                "item-type": "any"
+                "item-type": "composite-step"
+            }
+        },
+        "composite-step": {
+            "mapping": {
+                "properties": {
+                    "name": "string-steps-context",
+                    "id": "non-empty-string",
+                    "run": {
+                        "type": "string-steps-context",
+                        "required": true
+                    },
+                    "env": "step-env",
+                    "working-directory": "string-steps-context",
+                    "shell": "string-steps-context"
```

Comment:
Is shell required?

Reply:
@ericsciple and I talked about it and we decided that it's best to leave it optional. If the user doesn't put anything there, in ScriptHandler, it will default to the runner's OS default shell.

Reply:
Also, this follows the same consistency and logic as how e do it in the workflow so I think it makes sense.

Reply:
Thinking about this a bit more... let's make it require (and `non-empty-string`).

I think during the original ADR review with product we said required. Rationale because it's different than adding a `run` step to your own workflow where you control the OS. In this case, action-author doesn't control the OS. Should be explicit to avoid unexpected behavior.

Reply:
Yup, that makes sense. 

(Reminder to self: will need to update all documentation to have required shell attribute for each run step)